providers.tf

```diff
@@ -0,0 +1,10 @@
+# Copyright © 2022, Oracle and/or its affiliates.
+# All rights reserved. Licensed under the Universal Permissive License (UPL), Version 1.0 as shown at https://oss.oracle.com/licenses/upl.
+
+provider "oci" {
+  tenancy_ocid     = var.tenancy_ocid
+  user_ocid        = var.user_ocid
+  fingerprint      = var.fingerprint
+  private_key_path = var.private_key
```

Comment:
Since this is using the private key contents directly as a variable, I thought this should be private_key rather than private_key_path. However, if it authenticates for you this way, then it should be fine.
https://docs.oracle.com/en-us/iaas/Content/API/SDKDocs/terraformproviderconfiguration.htm
`private_key = var.private_key`